starfish/pipeline/features/decoded_spots.py

```diff
@@ -0,0 +1,24 @@
+import pandas as pd
+from ._validated_table import ValidatedTable
+
+
+class DecodedSpots(ValidatedTable):
+
+    required_fields = {
+        # TODO ambrosejcarr change barcode -> codeword, gene -> gene_name (requires rewrite of codebook)
+        'barcode',  # the code word that corresponds to this gene
+        'quality',  # spot y-coordinate
+        'spot_id',  # integer spot id
+        'gene'  # string gene name
+    }
+
+    def __init__(self, decoded_spots):
+        """
+
+        Parameters
+        ----------
+        decoded_spots : pd.DataFrame
+
+        """
+        # call the validation routine, set self.data = required_fields
+        super().__init__(decoded_spots, self.required_fields)
```

Comment:
I would prefer `DecodedSpots.required_fields` to indicate that it's a class var.

Reply:
Good idea. Will fix.